### DISCUSSION TITLE
How does MCC support non-ASCII characters, Chinese characters are non-ASCII characters

### DISCUSSION THREAD

**Discussion by FIZxingchen** [ORelio/Minecraft-Console-Client#2555] [Q&A]
mcc怎么支持非ASCII字符的，中文字符就属于非ASCII字符,我想用机器人自动私信玩家，但是却不能私信中文玩家,需要改源代码吗(MCC how to support non-ASCII characters, Chinese characters are non-ASCII characters, I want to use robot automatic private message players, but can not private message Chinese players, need to change the source code)

**Comment by ReinforceZwei** [comment]
原生伺服器并不支持英文数字以外的玩家名称，因此MCC会过滤掉不符合规则的名字。你需要自行修改源代码来删除过滤功能。我们可能会考虑新增设定来关掉过滤，但我们需要先讨论一下。

Vanilla server support player name with alphabet and number only. MCC will filter out invalid player name. You have to modify the source code to remove this behavior. We might consider adding a switch to disable filtering invalid player name but we need further discussion.

**Reply by FIZxingchen** [reply]
谢谢你的指导。我也希望能添加这个功能，那么，我应该怎么修改源代码呢，是在Settings.cs中吗(Thank you for your guidance. I wish I could add this feature, so how do I modify the source code, in Settings.cs?)

**Reply by FIZxingchen** [reply]
C:\Users\Administrator.DC-20210409OLET\Desktop\Minecraft-Console-Client-20230716-226>dotnet build
MSBuild version 17.6.8+c70978d4d for .NET
MSBUILD : error MSB1011: 此文件夹中包含多个项目或解决方案文件，请指定要使用的项目或解决方案文件(why?)

**Reply by ReinforceZwei** [reply]
See https://github.com/MCCTeam/Minecraft-Console-Client/issues/2517#issuecomment-1574640759
For building from source, see https://mccteam.github.io/guide/installation.html#building-from-the-source-code

**Reply by FIZxingchen** [reply]
$ git clone https://github.com/MCCTeam/Minecraft-Console-Client.git --recursive
Cloning into 'Minecraft-Console-Client'...
fatal: unable to access 'https://github.com/MCCTeam/Minecraft-Console-Client.git/': Recv failure: Connection was reset

**Reply by ReinforceZwei** [reply]
这是你的网络问题，你要自己想办法

**Reply by FIZxingchen** [reply]
好的，我会想办法,克隆好后应修改哪个文件来支持中文呢，感谢你用中文与我交流(Okay, I'll figure out which file to modify to support Chinese after cloning, thank you for communicating with me with Chinese)

**Comment by ReinforceZwei** [comment]
将这个留言https://github.com/MCCTeam/Minecraft-Console-Client/issues/2517#issuecomment-1574640759 提到的那几行代码删除。（[McClient.cs](https://github.com/MCCTeam/Minecraft-Console-Client/blob/9855e2e0f19942006bb62ec230bc6a935b5da270/MinecraftClient/McClient.cs#L2953-L2955)）

Delete the code mentioned in the comment.

**Reply by FIZxingchen** [reply]
我找到了McClient.cs,是让我删除第2517行的代码吗
(I found McClient.cs. Is this code for me to delete line 2517)

**Reply by ReinforceZwei** [reply]
2953-2955行

**Reply by FIZxingchen** [reply]
谢谢, thank you

**Reply by FIZxingchen** [reply]
Is /// <summary>
        /// Set client player's ID for later receiving player's own properties???，我把空格算上去了(I put in the spaces)

**Comment by FIZxingchen** [comment]
你能告诉我是什么代码吗，我自己找(Can you tell me what the code is? I'll find it myself)

**Reply by ReinforceZwei** [reply]
![image](https://github.com/MCCTeam/Minecraft-Console-Client/assets/39955851/0b08124b-af23-43f4-a30a-4385e1a12981)
黄色那三行

**Comment by FIZxingchen** [comment]
Thank you very much. It's settled,!!!!!!!!oh my ga

**Comment by FIZxingchen** [comment]
还有一个问题，我按照方法弄翻译中文，但是不成功(There is another problem. I tried to translate the Chinese according to the method, but it was not successful)如果有时间的话教一下我(Teach me if you have time)^*^

**Reply by ReinforceZwei** [reply]
翻译是什么意思？如果是说mcc显示语言的话自己编译的只有英文。
For self-compiled MCC, only English is available.

**Comment by FIZxingchen** [comment]
![ZH156CF%Y HEK2VTQY4LNT8](https://github.com/MCCTeam/Minecraft-Console-Client/assets/140962504/3d316ff8-ef2c-4bbc-a185-695000695339)
这个

**Reply by ReinforceZwei** [reply]
最新版本新增了`IgnoreInvalidPlayerName`设定

https://github.com/MCCTeam/Minecraft-Console-Client/releases/tag/20230802-227
